ls1intum/Artemis
# `Tutorial groups`: Fix various bugs concerning the tutorial group feature

### Checklist
#### General
#### Server
#### Client

### Motivation and Context
- Bug #6037 
- Bug #6036
- Bug #6035
- Feature Request #6039

### Description
- Closes #6037 and #6036 by allowing the user to edit the global configuration in the tutorial group checklist view when activating the tutorial groups feature for the first time in commit 8028831cf9f78d8af4e7e047dd48a5cfdd336716
  - Note: Now we require first the time zone for the course to be defined before a tutorial group configuration can be defined. When the configuration is then defined it will immediately route you to the tutorial group management page.If you want to try changing it from the checklist page, navigate back via the back button in the browser or open the checklist page manually again by entering `course-management/{courseId}/tutorial-groups-checklist`
- Closes #6039 by allowing users to define the language as a string basically allowing all the languages they want. I also implemented an autocomplete of existing languages used in other tutorial groups of the course. 3547f5915c848ad772d23492460a14e40a8dba14
- Closes #6035 date picker is now correctly positioned next to the input field (see screenshot). 2221796790ccfcf090bc0249e838ff5107a0c7d5

### Steps for Testing
- For the first fix, you could test the following scenarios:
  - When the user navigates back to the tutorial group checklist page via browser back button or manual URL entry, they should be able to edit the global configuration again.

- For the second fix, you could test the following scenarios:
  - When the user creates or edits a tutorial group, they should be able to define any language as a string in a text input field.
  - When the user types in a language that is already used in other tutorial groups of the course, they should see an autocomplete suggestion with that language name. (see screenshot)

- For the third fix, you could test the following scenarios:
  - The date picker should not overlap with any other elements on the page and should be aligned with the input field (see screenshot)

### Review Progress
#### Code Review
#### Manual Tests

### Test Coverage





### Screenshots
<img width="1576" alt="image" src="https://user-images.githubusercontent.com/29718932/226734159-a5c09be8-cd87-4382-9cd1-13da99105ab9.png">
<img width="1025" alt="image" src="https://user-images.githubusercontent.com/29718932/226744560-c0091040-62a7-4458-9add-e6b70ba89144.png">

Video of Bugfixes for 
- Bug #6037 
- Bug #6036
- Bug #6035

https://user-images.githubusercontent.com/29718932/227650621-9d18b891-4d77-44b6-9fea-b9f382ce0957.mp4


- Feature Request #6039

https://user-images.githubusercontent.com/29718932/227654418-72adcc97-87c0-46de-8860-b11c40420218.mp4
